--- zxing-cpp-e0e40ddec63f38405aca5c8c1ff60b85ec8b1f10/CMakeLists.txt.orig	2021-06-19 09:30:02.182786429 +0200
+++ zxing-cpp-e0e40ddec63f38405aca5c8c1ff60b85ec8b1f10/CMakeLists.txt	2021-06-19 09:34:19.041394907 +0200
@@ -3,6 +3,7 @@
 project(zxing)
 
 option(BUILD_TESTING "Enable generation of test targets" OFF)
+option(USE_OPENCV "OpenCV interface" ON)
 
 set(CMAKE_LIBRARY_PATH /opt/local/lib ${CMAKE_LIBRARY_PATH})
 
@@ -57,6 +58,7 @@
 endif()
 
 # OpenCV classes
+if(USE_OPENCV)
 find_package(OpenCV)
 if(OpenCV_FOUND)
     file(GLOB_RECURSE LIBZXING_OPENCV_FILES
@@ -87,6 +89,7 @@
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
+endif()
 
 # Add cli executable.
 file(GLOB_RECURSE ZXING_FILES
