--- zxing-cpp-6b3cbe02a332bff0f5ba0416f221d3d3876afdc2/CMakeLists.txt.orig	2016-11-22 04:34:19.000000000 +0100
+++ zxing-cpp-6b3cbe02a332bff0f5ba0416f221d3d3876afdc2/CMakeLists.txt	2017-04-16 21:29:53.022165125 +0200
@@ -65,7 +65,7 @@
 
 include_directories("./core/src/")
 add_library(libzxing ${LIBZXING_FILES})
-set_target_properties(libzxing PROPERTIES PREFIX "")
+set_target_properties(libzxing PROPERTIES PREFIX "" SOVERSION 0)
 
 find_package(Iconv)
 if(ICONV_FOUND)
@@ -94,9 +94,9 @@
 add_executable(zxing ${ZXING_FILES})
 target_link_libraries(zxing libzxing)
 install(TARGETS zxing libzxing
-	LIBRARY DESTINATION lib
+	LIBRARY DESTINATION lib${LIB_SUFFIX}
 	RUNTIME DESTINATION bin
-	ARCHIVE DESTINATION lib)
+	ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(DIRECTORY core/src/zxing/ DESTINATION include/zxing FILES_MATCHING PATTERN "*.h")
 
 # Add testrunner executable.
